packages/python-google-compute-engine/google_compute_engine/networking/network_setup/network_setup.py

```diff
@@ -43,23 +43,45 @@ def __init__(self, dhclient_script=None, dhcp_command=None, debug=False):
     self.logger = logger.Logger(
         name='network-setup', debug=debug, facility=facility)
     self.distro_utils = distro_utils.Utils(debug=debug)
+    self.ipv6_initialized = False
 
   def EnableIpv6(self, interfaces):
     """Enable IPv6 on the list of network interfaces.
 
     Args:
       interfaces: list of string, the output device names for enabling IPv6.
     """
-    if not interfaces or set(interfaces) == self.ipv6_interfaces:
+    if not interfaces or self.ipv6_interfaces == set(interfaces):
       return
 
     self.logger.info('Enabling IPv6 on Ethernet interface: %s.', interfaces)
-    self.ipv6_interfaces = set(interfaces)
+    self.ipv6_interfaces = self.ipv6_interfaces.union(set(interfaces))
+    self.ipv6_initialized = True
```

Comment:
is this safe to do before the call to utils.enableipv6? what if that failed?

Reply:
which call are you referring to ? btw any failures here are similar to IPv4 failures.

Reply:
Note that you have the same issue for IPv4 and I followed the existing pattern. yes, we could try fail but we don't raise the exception all the way up for the other APIs either.